store/heightsub.go

```diff
@@ -18,13 +18,13 @@ type heightSub[H header.Header[H]] struct {
 	// that has been fully verified and inserted into the subjective chain
 	height       atomic.Uint64
 	heightReqsLk sync.Mutex
-	heightReqs   map[uint64]map[chan H]struct{}
+	heightSubs   map[uint64]chan struct{}
```

Comment:
This has to be a map/set like before to support multiple waiters...

Try making two subscribers and the second one will override the first one...

Reply:
Made a test case that would fail here: https://github.com/celestiaorg/go-header/pull/242

Reply:
So the test case is not failing, as you can have a single channel serving multiple readers. However, current logic will fail on context canceling, so we need to either have a monotonic counter or make a chan per subscriber as before

Reply:
Done